wordpress-mobile/WordPress-Android
# [Jetpack Content Migration Flow] Implement Error Screen support

Fixes #17489

This PR adds support for showing the error screens (Networking & General) in the WordPress to Jetpack migration flow.

There's only one final type for the errors in `UiState.Error`, while not ideal this limitation was enforced by the need to show a processing state on these screens, and using distinct types for each error resulted in too much complications by having to cast each error type in order to change the `isProcessing` flag. Providing the specifics for each error type is done by instantiating the `Error` data class with the right values via 2 different methods. Ideally we would have 2 different error types but as explained this would result in over-complication of the code.

Adaptation of the header icon for dark mode is not handled, to be done together with adapting the the icons for all other migration flow screens in:
- https://github.com/wordpress-mobile/WordPress-Android/issues/17491


> 🏁 **All these changes are behind a temporary feature flag.**

To test:

 <details>
   <summary><b>Prerequisite</b> - local changes to show error screen</summary>

   <br>

   Replace `tryMigration()` in `JetpackMigrationViewModel.kt:73` to test errors:
   https://github.com/wordpress-mobile/WordPress-Android/blob/f77ec1e4c81773731f0be7d371c5d3b207c69e87/WordPress/src/main/java/org/wordpress/android/ui/main/jetpack/migration/JetpackMigrationViewModel.kt#L70-L75
   - Replace with `postGenericErrorState()` to test the **Generic Error**
   - Replace with `postNetworkingErrorState()` to test the **Networking Error**
---
</details>

0. Checkout the branch of this PR locally
1. Modify the locally checked out code to show one of the errors when `Continue` is tapped on the Welcome Screen — see **Prerequisite** collapsible panel above.
2. Open the Jetpack app & login
3. Toggle the temporary feature flag (this is only needed once)
   a. Go to `Me` → `App Settings` → `Debug Settings`
   b. Enable the `JetpackMigrationFlowFeatureConfig` & restart the app.
4. Tap the `Continue` button on the "Welcome to Jetpack!" screen
5. **Expect** to see one of the error screens (depending which one was added via local changes)
6. **Verify** that the error screen matches the design
7. Tap `Need help?`
8. **Expect** To see the Support Screen customized for the migration flow (see PR [#17449](https://github.com/wordpress-mobile/WordPress-Android/pull/17449))
9. Go back
10. Tap `Try Again` on the error screen
11. **Expect** to see a processing state simulating the migration progress
12. **Expect** to land on the "Notifications now come from Jetpack" screen

📣 Please test both errors (Generic & Networking), with each variation via local changes:
- `postGenericErrorState()`
- `postNetworkingErrorState()`

### Previews

| Generic Error | Networking Error |
| - | - |
| <img width="314" src="https://user-images.githubusercontent.com/4588074/202797006-3d32691f-9cb7-4903-8092-09740ea9c07f.png"> | <img width="314" src="https://user-images.githubusercontent.com/4588074/202797028-51547f39-b520-4fbd-8c7e-5eb1f2aedbe7.png"> |

## Regression Notes
1. Potential unintended areas of impact
   N/a - Changes are behind feature flag.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
   N/a

4. What automated tests I added (or what prevented me from doing so)
   N/a

PR submission checklist:
